Source/WebGPU/WGSL/AttributeValidator.cpp

```diff
@@ -451,10 +452,9 @@ void AttributeValidator::visit(AST::StructureMember& member)
                 continue;
             }
             auto alignmentValue = constantValue->integerValue();
-            auto isPowerOf2 = !(alignmentValue & (alignmentValue - 1));
             if (alignmentValue < 1)
                 error(attribute.span(), "@align value must be positive"_s);
-            else if (!isPowerOf2)
+            else if (!isPowerOfTwo(static_cast<uint64_t>(alignmentValue)))
```

Comment:
I am a little surprised a type cast is needed. Should we be using a function more like “make unsigned” instead of a cast to a specific type?

Reply:
Or maybe this?
```
isPowerOfTwo<std::make_unsigned_t<decltype(alignmentValue)>>(alignmentValue)
```

Reply:
I guess I’m thinking we’d make `unsigned_cast` or `make_unsigned_cast`, which would use `make_unsigned_t` in its implementation, but spell it `unsignedCast`. Like `byteCast` it would be a less powerful cast for a specific purpose, it would be `isPowerOfTwo(unsignedCast(alignmentValue))`.

Reply:
Ok. I can look into introducing such as casting function and adopt it here.

Reply:
Following up in https://github.com/WebKit/WebKit/pull/45095